lib/ansible/plugins/vars/host_group_vars.py

```diff
@@ -95,6 +95,10 @@ def get_vars(self, loader, path, entities, cache=True):
                     for found in found_files:
                         new_data = loader.load_from_file(found, cache=True, unsafe=True)
                         if new_data:  # ignore empty files
+                            try:
+                                validate_variable_names(new_data.keys())
+                            except TypeError as e:
+                                raise AnsibleParserError("Invalid variable name in '%s' specified: '%s'" % (found, to_native(e)))
```

Comment:
same here, instead of adding to the overridable plugin we might want to move this to consumers (vars manager/inventory cli)